pkg/scheduler/plugins/deviceshare/allocator_gpu.go

```diff
@@ -114,6 +122,63 @@ func removeZeroDevice(originalResources deviceResources) deviceResources {
 	return refinedResources
 }
 
+func (a *GPUAllocator) generalAllocate(requestCtx *requestContext, nodeDevice *nodeDevice, desiredCount int, maxDesiredCount int, allocateContext *AllocateContext) ([]*apiext.DeviceAllocation, *framework.Status) {
+	allocations, status := allocateByDeviceTopology(requestCtx.gpuRequirements, nodeDevice.gpuTopologyScope, allocateContext)
+	if !status.IsSuccess() {
+		return nil, status
+	}
+	// if same NUMANode or PCIE are not required, allocateByDeviceTopology may return (nil, nil), so we should check if allocations is nil
+	if len(allocations) != 0 {
+		return allocations, nil
+	}
+
+	return defaultAllocateDevices(nodeDevice, requestCtx, requestCtx.gpuRequirements.requestsPerGPU, desiredCount, maxDesiredCount, schedulingv1alpha1.GPU, nil)
+}
+
+func (a *GPUAllocator) allocateByTemplate(requestCtx *requestContext, nodeDevice *nodeDevice, desiredCount int, maxDesiredCount int, allocateContext *AllocateContext) ([]*apiext.DeviceAllocation, *framework.Status) {
+	if !requestCtx.gpuRequirements.gpuShared {
+		return nil, nil
+	}
+	vendor, model := requestCtx.node.Labels[apiext.LabelGPUVendor], requestCtx.node.Labels[apiext.LabelGPUModel]
+	if vendor != apiext.GPUVendorHuawei {
```

Comment:
why this is hardcoded?

Reply:
Because we want to schedule by template only for vendors which enforce it, if we don't filter here, this would cause other irrelevant vendors meet the unexpected `no matched GPU shared resource template` error.

Reply:
What I mean is whether the information of "which manufacturers need to be allocated according to the template" can be configured